probe/probe.go

```diff
@@ -0,0 +1,315 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of KubeArmor
+
+package probe
+
+
+import (
+	"fmt"
+	"syscall"
+    "strings"
+	"golang.org/x/mod/semver"
+    "log"
+    "github.com/fatih/color"
+    "os"
+    "io"
+    "time"
+    "github.com/kubearmor/kubearmor-client/k8s"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"github.com/kubearmor/kubearmor-client/deployment"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var white = color.New(color.FgWhite)
+var boldWhite = white.Add(color.Bold)
+var karmorprobe = "karmor-probe"
+
+
+
+// Options for probe daemonset options install
+type ProbeOptions struct {
+    Namespace      string
+    ProbeDaemonImage string
+    Full          bool
+}
+
+// K8sInstaller for karmor install
+func probeDaemonInstaller(c *k8s.Client, o ProbeOptions) error {
+    
+    daemonset := deployment.GenerateDaemonSet(o.Namespace)
+    daemonset.Spec.Template.Spec.Containers[0].Image = o.ProbeDaemonImage
+    if _, err := c.K8sClientset.AppsV1().DaemonSets(o.Namespace).Create(context.Background(), daemonset, metav1.CreateOptions{}); err != nil {
+        if !strings.Contains(err.Error(), "already exists") {
+            return err
+        }
+    }
+    return nil
+
+
+}
+
+
+func probeDaemonUninstaller(c *k8s.Client, o ProbeOptions) error {
+    color.Yellow("\t Deleting Karmor Probe DaemonSet ...\n")
+	if err := c.K8sClientset.AppsV1().DaemonSets(o.Namespace).Delete(context.Background(), karmorprobe, metav1.DeleteOptions{}); err != nil {
+		if !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+		fmt.Print("Karmor probe DaemonSet not found ...\n")
+	}
+
+
+	return nil
+}
+
+func PrintProbeResult(c *k8s.Client, o ProbeOptions) error{
+    if o.Full {
+        checkHostAuditSupport()
+        checkLsmSupport(getHostSupportedLSM())
+
+        if err := probeDaemonInstaller(c, o); err != nil {
+            return err
+        }
+        color.Yellow("\t Creating probe daemonset ...")
+        time.Sleep(2 * time.Second)
+        probeNode(c)
+        if err := probeDaemonUninstaller(c, o); err != nil {
+            return err
+        }
+	}else{
+        checkHostAuditSupport()
+        checkLsmSupport(getHostSupportedLSM())
+        color.Blue("To get full probe, a daemonset will be deployed in your cluster - This daemonset will be deleted after probing")
+        color.Blue("Use --full tag to get full probing")
+    }
+    return nil
+}
+
+
+func checkLsmSupport(supportedLSM string) {
+    fmt.Printf("\t Enforcement:") 
+    if strings.Contains(supportedLSM, "selinux"){
+        color.Yellow(" Partial (Supported LSMs: "+ supportedLSM + ") \n\t To have full enforcement support, apparmor must be supported")
+    }else if strings.Contains(supportedLSM, "apparmor"){
```

Comment:
```suggestion
    }else if strings.Contains(supportedLSM, "apparmor") || strings.Contains(supportedLSM, "bpf") {
```

Since we recently added support for BPF LSM as part of v0.5 release.